From: Rainer Gerhards <rgerhards@adiscon.com>
Date: Wed, 8 Jul 2015 08:51:59 +0200
Subject: bugfix ompgsql: transaction were improperly handled

now transaction support is solidly disabled until we have enough requests
to implement it again. Module still works fine in single insert mode.

closes https://github.com/rsyslog/rsyslog/issues/399

(cherry-picked from commit 73c9a28653d4e92a6c71210232a83b4b72cdf778)
---
 plugins/ompgsql/ompgsql.c | 11 +++++++++--
 1 file changed, 9 insertions(+), 2 deletions(-)

diff --git a/plugins/ompgsql/ompgsql.c b/plugins/ompgsql/ompgsql.c
index 6b89717..72b49ee 100644
--- a/plugins/ompgsql/ompgsql.c
+++ b/plugins/ompgsql/ompgsql.c
@@ -234,6 +234,7 @@ writePgSQL(uchar *psz, instanceData *pData)
 	bHadError = tryExec(psz, pData); /* try insert */
 
 	if(bHadError || (PQstatus(pData->f_hpgsql) != CONNECTION_OK)) {
+#if 0		/* re-enable once we have transaction support */
 		/* error occured, try to re-init connection and retry */
 		int inTransaction = 0;
 		if(pData->f_hpgsql != NULL) {
@@ -242,7 +243,9 @@ writePgSQL(uchar *psz, instanceData *pData)
 				inTransaction = 1;
 			}
 		}
-		if ( inTransaction == 0 ) {
+		if ( inTransaction == 0 )
+#endif
+		{
 			closePgSQL(pData); /* close the current handle */
 			CHKiRet(initPgSQL(pData, 0)); /* try to re-open */
 			bHadError = tryExec(psz, pData); /* retry */
@@ -282,6 +285,7 @@ CODESTARTtryResume
 ENDtryResume
 
 
+#if 0 /* re-enable when TX support is added again */
 BEGINbeginTransaction
 CODESTARTbeginTransaction
 	dbgprintf("ompgsql: beginTransaction\n");
@@ -289,6 +293,7 @@ CODESTARTbeginTransaction
 	       initPgSQL(pWrkrData->pData, 0);
 	iRet = writePgSQL((uchar*) "begin", pWrkrData->pData); /* TODO: make user-configurable */
 ENDbeginTransaction
+#endif
 
 
 BEGINdoAction
@@ -303,11 +308,13 @@ finalize_it:
 ENDdoAction
 
 
+#if 0 /* re-enable when TX support is added again */
 BEGINendTransaction
 CODESTARTendTransaction
 	iRet = writePgSQL((uchar*) "commit;", pWrkrData->pData); /* TODO: make user-configurable */
 dbgprintf("ompgsql: endTransaction\n");
 ENDendTransaction
+#endif
 
 
 BEGINparseSelectorAct
@@ -386,7 +393,7 @@ BEGINqueryEtryPt
 CODESTARTqueryEtryPt
 CODEqueryEtryPt_STD_OMOD_QUERIES
 CODEqueryEtryPt_STD_OMOD8_QUERIES
-CODEqueryEtryPt_TXIF_OMOD_QUERIES /* we support the transactional interface! */
+/* CODEqueryEtryPt_TXIF_OMOD_QUERIES currently no TX support! */ /* we support the transactional interface! */
 ENDqueryEtryPt
 
 
